accord-core/src/main/java/accord/api/Write.java

```diff
@@ -30,5 +31,5 @@
  */
 public interface Write
 {
-    AsyncChain<Void> apply(Seekable key, SafeCommandStore safeStore, Timestamp executeAt, DataStore store);
+    AsyncChain<Void> apply(Seekable key, SafeCommandStore safeStore, Timestamp executeAt, DataStore store, Command.Executed command);
```

Comment:
I kinda don't like exposing `Command` in the API and kinda prefer `TxnId`... you do the following in C*

```
      // Pull updates for this key that did not require completion form the transaction itself and apply them: 
            TxnUpdate txnUpdate = (TxnUpdate) command.partialTxn().update();
            assert txnUpdate != null : "PartialTxn should contain an update if we're applying a write!";
            List<Update> updates = txnUpdate.completeUpdatesForKey((RoutableKey) key);
            updates.forEach(update -> results.add(update.write(timestamp, nowInSeconds)));
```

so you will need it, but I just kinda don't like it...

Reply:
@belliottsmith above suggests using just the `PartialTxn` or the `Update`. You're suggesting perhaps passing `TxnId` and getting the command from `SafeCommandStore` (along w/ the key, which I think we also have available)?